java/src/jmri/jmrit/logix/Engineer.java

```diff
@@ -40,16 +42,17 @@
     private boolean _waitForClear = false;  // waits for signals/occupancy/allocation to clear
     private boolean _waitForSync = false;  // waits for train to catch up to commands
     private boolean _waitForSensor = false; // wait for sensor event
-    private boolean _runOnET = false;   // Execute commands on ET only - do not synch
+    private boolean _runOnET = false;   // Execute commands on ET only - do not synchn
     private boolean _setRunOnET = false; // Need to delay _runOnET from the block that set it
     private int _syncIdx;           // block order index of current command
     protected DccThrottle _throttle;
     private final Warrant _warrant;
     private final List<ThrottleSetting> _commands;
     private Sensor _waitSensor;
     private int _sensorWaitState;
-    private ThrottleRamp _ramp;
     final ReentrantLock _lock = new ReentrantLock(true);    // Ramp needs to block script speeds
+    private static Object _lockObject = new Object(); // used for synchronizing threads for _ramp
```

Comment:
It seems this means that if there are multiple Engineers, they all share the same synchronization lock, and is that really what’s wanted?

Reply:
I am actually not sure.  

Previously, the synchronization was using a thread instance, which shouldn’t be done.

At least one of those references was in a static method, hence turning this into a static....

But some of the tests are failing, so this probably needs to be rethought, as it isn’t right.

Reply:
This should be fixed now.  The static reference was on an intern class, soI have it a lock object of its own.